sklearn/cluster/_hdbscan/_reachability.pyx

```diff
@@ -1,41 +1,52 @@
-# mutual reachability distance compiutations
+# mutual reachability distance computations
 # Authors: Leland McInnes <leland.mcinnes@gmail.com>
 #          Meekail Zain <zainmeekail@gmail.com>
 # License: 3-clause BSD
 
 import numpy as np
 from scipy.sparse import issparse
-from ...neighbors import BallTree, KDTree
 
 cimport numpy as cnp
 from cython.parallel cimport prange
-from libc.math cimport isfinite
+from libc.math cimport isfinite, INFINITY
 
 
-def mutual_reachability(distance_matrix, min_points=5, max_dist=0.0):
-    """Compute the weighted adjacency matrix of the mutual reachability
-    graph of a distance matrix. Note that computation is performed in-place for
-    `distance_matrix`. If out-of-place computation is required, pass a copy to
-    this function.
+def mutual_reachability_graph(
+    distance_matrix, min_samples=5, max_distance=0.0, copy=False
+):
+    """Compute the weighted adjacency matrix of the mutual reachability graph.
+
+    The mutual reachability distance used to build the graph is defined as::
+
+        max(d_core(x_p), d_core(x_q), d(x_p, x_q))
+
+    and the core distance `d_core` is defined as the distance between a point
+    `x_p` and its k-th nearest neighbor.
 
     Parameters
     ----------
-    distance_matrix : ndarray or sparse matrix of shape (n_samples, n_samples)
+    distance_matrix : {ndarray, sparse matrix} of shape (n_samples, n_samples)
         Array of distances between samples. If sparse, the array must be in
         `LIL` format.
 
-    min_points : int, default=5
+    min_samples : int, default=5
         The number of points in a neighbourhood for a point to be considered
         a core point.
 
-    max_dist : float, default=0.0
+    max_distance : float, default=0.0
         The distance which `np.inf` is replaced with. When the true mutual-
         reachability distance is measured to be infinite, it is instead
-        truncated to `max_dist`.
+        truncated to `max_dist`. Only used when `distance_matrix` is a sparse
+        matrix.
+
+    copy : bool, default=False
```

Comment:
We can add the keyword instead of the telling in the documentation.
We still do not do in place for the sparse matrix thought.